### ISSUE TITLE
Game will not load

### ISSUE THREAD

**Issue by amourelius** [mmtrt/leagueoflegends#225]
Hello,
The game itself is not loading now that I am using rev. 148. The client/launcher is greatly improved following issue #223  instructions. I was able to play in the past then I couldn't get the client to open all but rarely it would after many attempts.

When I press "start game" testing in training mode, it counts down, client minimizes then nothing for a few seconds. Then the client refocuses and asks me to reconnect which continues the cycle. 

Attempts: `sudo snap remove --purge leagueoflegends` 
then  `snap install leagueoflegends --devmode --beta`
it asks me to run `snap install --candidate wine-platform-7-staging-core20` which i did.

Client/Launcher work so much better with this but now I can not get the game to load. 

Here is what I get starting the launcher and client: 
`wineserver: using server-side synchronization.
0078:err:wineusb:DriverEntry Failed to initialize Unix library, status 0xc0000135.
0078:err:ntoskrnl:ZwLoadDriver failed to create driver L"\\Registry\\Machine\\System\\CurrentControlSet\\Services\\wineusb": c0000135
01fc:err:winediag:getaddrinfo Failed to resolve your host name IP
0244:err:rpc:I_RpcGetBuffer no binding
02c4:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
0350:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
0350:err:module:import_dll Library api-ms-win-crt-string-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
0350:err:module:import_dll Library api-ms-win-crt-math-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
0350:err:module:import_dll Library api-ms-win-crt-runtime-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
0350:err:module:import_dll Library api-ms-win-crt-private-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
0350:err:module:import_dll Library api-ms-win-crt-time-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
0350:err:module:import_dll Library api-ms-win-crt-string-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
0350:err:module:import_dll Library api-ms-win-crt-math-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
0350:err:module:import_dll Library api-ms-win-crt-runtime-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
0350:err:module:import_dll Library api-ms-win-crt-private-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
0350:err:module:import_dll Library api-ms-win-crt-time-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
0350:err:module:import_dll Library api-ms-win-crt-string-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
0350:err:module:import_dll Library api-ms-win-crt-math-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
0350:err:module:import_dll Library api-ms-win-crt-runtime-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
0350:err:module:import_dll Library api-ms-win-crt-private-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
0350:err:module:import_dll Library api-ms-win-crt-time-l1-1-0.dll (which is needed by L"C:\\Riot Games\\Riot Client\\UX\\D3DCompiler_47.dll") not found
03cc:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems`

Then output when attempting to "start game" : 

`0bb4:err:virtual:virtual_setup_exception stack overflow 836 bytes in thread 0bb4 addr (nil) stack 0x3710cbc (0x3710000-0x3711000-0x3810000)
0bc4:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
0ad4:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
0bb0:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
015c:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
09e0:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
0b84:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
0104:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
0080:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
0a40:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
0b04:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
0b08:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
0a6c:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems`

I attempted to search around and found this may be wine 7.4 related issue that was remedied in 7.5 but honestly have no clue. 
https://forum.winehq.org/viewtopic.php?t=36337

Thank you

Ubuntu 20.04 
snap list: 
![image](https://user-images.githubusercontent.com/127454603/228658096-dccfddaa-5a2e-4e00-8048-a7ebefd6c1b3.png)

**Comment by mmtrt** [comment]
that looks like broken game install try again.

**Comment by mmtrt** [comment]
Try installing d3dcompiler_47

`leagueoflegends.winetricks d3dcompiler_47`

**Comment by amourelius** [comment]
d3dcompiler_47 was already installed.

ran the next one and it failed with ~3600 lines of error

`------------------------------------------------------
warning: Your version of wine 4.11-21066-g6703dc23fbb is no longer supported upstream. You should upgrade to 7.x
------------------------------------------------------
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Using winetricks 20230212-next - sha256sum: 6a36f1911ee48c48978f12a3a0dea33cb871052fcce6910d6e1b664dd9090ac9 with wine-4.11-21066-g6703dc23fbb (Staging) and WINEARCH=win64
Executing w_do_call vcrun2015
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Executing load_vcrun2015 
Executing mkdir -p /home/metamask/snap/leagueoflegends/common/.cache/winetricks/vcrun2015
Executing cd /home/metamask/snap/leagueoflegends/common/.cache/winetricks/vcrun2015
Downloading https://download.microsoft.com/download/6/D/F/6DF3FF94-F7F9-4F0B-838C-A328D1A7D0EE/vc_redist.x86.exe to /home/metamask/snap/leagueoflegends/common/.cache/winetricks/vcrun2015
--2023-03-29 19:04:28--  https://download.microsoft.com/download/6/D/F/6DF3FF94-F7F9-4F0B-838C-A328D1A7D0EE/vc_redist.x86.exe
Resolving download.microsoft.com (download.microsoft.com)... 69.192.209.23, 2600:1402:1400:28d::317f, 2600:1402:1400:289::317f
Connecting to download.microsoft.com (download.microsoft.com)|69.192.209.23|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 14458272 (14M) [application/octet-stream]
Saving to: ‘vc_redist.x86.exe’

vc_redist.x86.exe                         100%[===================================================================================>]  13.79M  24.5MB/s    in 0.6s    

2023-03-29 19:04:29 (24.5 MB/s) - ‘vc_redist.x86.exe’ saved [14458272/14458272]

Executing cd /home/metamask
Using native,builtin override for following DLLs: api-ms-win-crt-private-l1-1-0 api-ms-win-crt-conio-l1-1-0 api-ms-win-crt-convert-l1-1-0 api-ms-win-crt-environment-l1-1-0 api-ms-win-crt-filesystem-l1-1-0 api-ms-win-crt-heap-l1-1-0 api-ms-win-crt-locale-l1-1-0 api-ms-win-crt-math-l1-1-0 api-ms-win-crt-multibyte-l1-1-0 api-ms-win-crt-process-l1-1-0 api-ms-win-crt-runtime-l1-1-0 api-ms-win-crt-stdio-l1-1-0 api-ms-win-crt-string-l1-1-0 api-ms-win-crt-utility-l1-1-0 api-ms-win-crt-time-l1-1-0 atl140 concrt140 msvcp140 msvcp140_1 msvcp140_atomic_wait ucrtbase vcomp140 vccorlib140 vcruntime140 vcruntime140_1
Executing wine C:\windows\syswow64\regedit.exe /S C:\windows\Temp\override-dll.reg
Executing wine64 C:\windows\regedit.exe /S C:\windows\Temp\override-dll.reg
------------------------------------------------------
warning: Working around wine bug 50894 -- Working around failing wusa.exe lookup via C:\windows\SysNative
------------------------------------------------------
Executing wine winecfg -v winxp64
------------------------------------------------------
warning: Running /snap/leagueoflegends/148/wine-platform/wine-staging/bin/wineserver -w. This will hang until all wine processes in prefix=/home/metamask/snap/leagueoflegends/common/.wine terminate
------------------------------------------------------
Executing cabextract -q --directory=/home/metamask/snap/leagueoflegends/common/.wine/dosdevices/c:/windows/temp/win32 /home/metamask/snap/leagueoflegends/common/.cache/winetricks/vcrun2015/vc_redist.x86.exe -F a10
Executing cabextract -q --directory=/home/metamask/snap/leagueoflegends/common/.wine/dosdevices/c:/windows/syswow64 /home/metamask/snap/leagueoflegends/common/.wine/dosdevices/c:/windows/temp/win32/a10 -F ucrtbase.dll
/home/metamask/snap/leagueoflegends/common/.wine/dosdevices/c:/windows/temp/win32/a10: WARNING; possible 16128 extra bytes at end of file.
Executing cd /home/metamask/snap/leagueoflegends/common/.cache/winetricks/vcrun2015
Executing wine vc_redist.x86.exe /q
wineserver: using server-side synchronization.`


then this hundreds of times:

`0078:err:wineusb:DriverEntry Failed to initialize Unix library, status 0xc0000135.
0078:err:ntoskrnl:ZwLoadDriver failed to create driver L"\\Registry\\Machine\\System\\CurrentControlSet\\Services\\wineusb": c0000135
0024:err:module:import_dll Library api-ms-win-core-string-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-errorhandling-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-timezone-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-file-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-namedpipe-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-handle-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-file-l2-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-heap-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-libraryloader-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-synch-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-processthreads-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-processenvironment-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-datetime-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-localization-l1-2-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-sysinfo-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-synch-l1-2-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-console-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-debug-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-processthreads-l1-1-1.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-file-l1-2-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-profile-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-memory-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
0024:err:module:import_dll Library api-ms-win-core-util-l1-1-0.dll (which is needed by L"C:\\windows\\syswow64\\ucrtbase.dll") not found
`

**Comment by mmtrt** [comment]
k I'll try to reproduce issue though game will break again next month patch anyways

**Comment by amourelius** [comment]
As a follow up, I have a second PC with the same version of Ubuntu 20.04 installed. 
I was able to duplicate this issue exactly on it as well.
I then tried upgrading my Ubuntu to 22.04, which made no difference in this error. Same issue after doing a clean install so I'm not sure what to think of this. I do have it running the --edge version, just painful with the client load issue but it will eventually work.

Either way im probably the 1 person in the world with this problem, and I imagine the game is going to break with the upcoming patch anyway doing away with 32 bit support. 

I hope that maybe if you get it working after the patch somehow it will resolve my issues, we'll see. Thanks again.
